### PR TITLE
Improve unit testing test suite finish performance

### PR DESCRIPTION
**Pull request by Therzok**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/635510

### CODE REVIEW

**Comment by Therzok** [comment]
Cc @slluis

File: main/src/addins/MonoDevelop.UnitTesting/Services/UnitTest.cs [hunk 1]
```diff
@@ -417,6 +419,9 @@ protected virtual void OnTestChanged ()
 		
 		protected virtual void OnTestStatusChanged ()
 		{
+			if (parent is UnitTestGroup) {
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.UnitTesting/Services/UnitTest.cs]: Haven't yet tracked down why this is needed. I think the test group is signalled to be done before its contained tests, so the status is not right

**Comment by Therzok** [comment]
```
2018-07-20T19:34:04.5460390Z 2018-07-20 19:34:04.544 vstool[1069:9676] warning: Can't find custom attr constructor image: /Users/vsts/agent/2.136.1/work/r1/a/VS for Mac/drop/tests/buildtemplates/packages/Xamarin.Android.Support.Annotations.27.0.2/lib/MonoAndroid81/Xamarin.Android.Support.Annotations.dll mtoken: 0x0a00000f due to: Cannot resolve dependency to assembly because it has not been preloaded. When using the ReflectionOnly APIs, dependent assemblies must be pre-loaded or loaded on demand through the ReflectionOnlyAssemblyResolve event. assembly:Mono.Android, Version=0.0.0.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065 type:<unknown type> member:(null) signature:<none>
2018-07-20T19:34:05.5134080Z Resource folder /Users/vsts/agent/2.136.1/work/r1/a/VS for Mac/drop/tests/buildtemplates/Xamarin.Android.FSharp.AndroidApp
2018-07-20T19:34:05.5151980Z Resource file name /Users/vsts/agent/2.136.1/work/r1/a/VS for Mac/drop/tests/buildtemplates/Xamarin.Android.FSharp.AndroidApp/Resources/Resource.designer.cs
```

**Comment by Therzok** [comment]
cc @nosami

File: main/src/addins/MonoDevelop.UnitTesting/Services/UnitTest.cs [hunk 2]
```diff
@@ -196,9 +196,11 @@ public virtual int CountTestCases ()
 		public TestStatus Status {
 			get { return status; }
 			set {
+				if (status == value)
+					return;
+
 				status = value;
 				OnTestStatusChanged ();
-				(Parent as UnitTestGroup)?.UpdateStatusFromChildren ();
 			}
```
Comment (mrward) [inline main/src/addins/MonoDevelop.UnitTesting/Services/UnitTest.cs]: UpdateStatusFromChildren is never called anywhere now so can it be removed? The new code now calls UnitTestGroup.OnTestStatusChanged which is different to UpdateStatusFromChildren.

**Comment by sgmunn** [comment]
@mrward can you approve / verify this so it can be merged

File: main/src/addins/MonoDevelop.UnitTesting/Services/UnitTest.cs (hunk 2 above)
Comment (Therzok) [reply main/src/addins/MonoDevelop.UnitTesting/Services/UnitTest.cs]: It is removed.

File: main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestGroup.cs [hunk 3]
```diff
@@ -110,21 +112,19 @@ static UnitTestResult GetLastResultDynamicaly (IEnumerable<UnitTest> tests)
 			return result;
 		}
 
-
-		internal void UpdateStatusFromChildren ()
```
Comment (Therzok) [inline main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestGroup.cs]: See here, it's removed.

Comment (mrward) [reply main/src/addins/MonoDevelop.UnitTesting/Services/UnitTestGroup.cs]: The code looks good. Was going to give it a quick test at some point....

**Comment by Therzok** [comment]
My test case was generating 1000 tests via a console app. I can share the project to show the difference.

**Pull request merged**